client/src/test/java/io/pravega/client/segment/impl/SegmentOutputStreamTest.java

```diff
@@ -776,39 +772,34 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
     public void testNoSuchSegment() throws Exception {
         UUID cid = UUID.randomUUID();
         PravegaNodeUri uri = new PravegaNodeUri("endpoint", SERVICE_PORT);
-
         MockConnectionFactoryImpl cf = new MockConnectionFactoryImpl();
-        ScheduledExecutorService executor = mock(ScheduledExecutorService.class);
-        implementAsDirectExecutor(executor); // Ensure task submitted to executor is run inline.
-        cf.setExecutor(executor);
-
+        @Cleanup("shutdown")
```

Comment:
There is a unit test base class to handle async unit tests which takes care of all the creation and cleanup of Thread Pools.

Have this test class extend from `ThreadPooledTestSuite`, override the `getThreadPoolSize` and have it return 0 (I believe that is the default). Then use `executorService()` to access the Thread Pool.